akka-actor/src/main/scala/akka/actor/AbstractProps.scala

```diff
@@ -20,9 +20,17 @@ private[akka] trait AbstractProps {
   /**
    * INTERNAL API
    */
-  private[akka] def validate(clazz: Class[_]) =
+  private[akka] def validate(clazz: Class[_]): Unit = {
     if (Modifier.isAbstract(clazz.getModifiers))
       throw new IllegalArgumentException(s"Actor class [${clazz.getName}] must not be abstract")
+    clazz match {
+      case _ if classOf[Actor].isAssignableFrom(clazz)                 =>
+      case _ if classOf[IndirectActorProducer].isAssignableFrom(clazz) =>
+      case _ =>
+        throw new IllegalArgumentException(
+          s"Actor class [${clazz.getName}] must be subClass of akka.actor.Actor or akka.actor.IndirectActorProducer.")
+    }
```

Comment:
Make it an if else instead since it doesn't actually pattern match anything

Reply:
yeap